blockchain/peer.go

```diff
@@ -0,0 +1,130 @@
+package blockchain
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	flow "github.com/tendermint/tendermint/libs/flowrate"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/types"
+)
+
+const defaultPeerTimeout = 15 * time.Second
+
+//--------
+// Peer
+var (
```

Comment:
we should try to avoid mutable global state. `peerTimeout` can be a config variable passed during initialization.